src/Renderer/Simulator/NumberHelpers.fs

```diff
@@ -143,11 +143,13 @@ let popupTextToFloat (popupUnitSuffix: string option) (text:string) =
 
 
 
+
+
 let floatValueToText (value:float) :string =
-    let asStr = string value
+    let asStr = if ((string value) |> Seq.exists (fun c -> c = 'e' || c = 'E')) then sprintf "%.10f" value else string value

```

Comment:
this function should be rewritten from scratch. It can be made more concise and at the same time do all the SI multipliers properly. It requires care, because if say always rounding to 4 sig figs 0.99999 should be displayed as 1, not 999.9m. The correct algorithm is:

1. normalise the number to m * 10^e  (where m lies in range 1000 <= m < 9999.5)
2. convert m to an int. If m = 10000 due to rounding errors set it equal to 9999
3. take m,e and from them work out the correct si multiplier so that the number m' is in the range 1 < m' < 999.9
4. ideally the precision (4 digits here) should be adjustable as a constant somewhere.